simpleclient_spring_hibernate/src/main/java/io/prometheus/client/spring/hibernate/SimpleClientHibernateApplicationListener.java

```diff
@@ -0,0 +1,48 @@
+package io.prometheus.client.spring.hibernate;
+
+import io.prometheus.client.hibernate.HibernateStatisticsCollector;
+import org.hibernate.Session;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.EntityManager;
+
+@Component
+public class SimpleClientHibernateApplicationListener implements ApplicationListener<ApplicationReadyEvent> {
+  @Autowired
+  private EntityManager entityManager;
+
+  @Value("${spring.application.name:bootstrap}")
+  private String appName = "bootstrap";
+
+  private HibernateStatisticsCollector collector = new HibernateStatisticsCollector();
+
+  public SimpleClientHibernateApplicationListener() {
+
+  }
+
+  public SimpleClientHibernateApplicationListener(
+      HibernateStatisticsCollector collector,
+      EntityManager entityManager,
+      String appName
+  ) {
+    this.collector = collector;
+    this.entityManager = entityManager;
+    this.appName = appName;
+  }
+
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent event) {
+    if (entityManager == null) {
+      return;
+    }
+
+    Session session = (Session) entityManager.getDelegate();
+
+    collector.add(session.getSessionFactory(), appName);
+    collector.register();
```

Comment:
called more than once by SpringBoot? not that i've seen, but i'm new to SpringBoot.

i moved collector initialization into the field declaration so that i could inject a mock for the sake of coverage. i'd be happy to replace the mock-heavy test with an application integration test which verifies collector configuration some other way.